build/scripts/validate-extension-metadata.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import groovy.io.FileType
+import groovy.json.JsonSlurper
+
+final String quarkusExtensionJsonRelPath = 'runtime/src/main/resources/META-INF/quarkus-extension.json'
+final List<String> messages = []
+project.basedir.eachFile FileType.DIRECTORIES, { extensionParentDir ->
+    if (new File(extensionParentDir, 'runtime/pom.xml').exists()) {
+        final File extensionJsonFile = new File(extensionParentDir, quarkusExtensionJsonRelPath)
+        final String shortName = extensionParentDir.getName()
+        final String shortPath = shortName + '/' + quarkusExtensionJsonRelPath
+        final boolean ancillary = (shortName.startsWith('core') || shortName.endsWith('-common'))
+        if (!extensionJsonFile.exists()) {
+            messages.add(shortPath + ' is missing')
+        } else if (!ancillary) {
```

Comment:
Actually not sure we want to ignore all these checks in the ancillary extensions. WDYT @lburgazzoli ? Maybe they should have a name, empty labels (or maybe integration, internal ?) and the guide does not matter?

Reply:
core-* do already have the extension metadata file (well core-cloud has an empty one but that's a mistake) and probably xml-common should now be renamed core-xml so yes, every non support extension should have proper  metdata

Reply:
Thanks. One more question: do we still consider the extensions satisfying `shortName.startsWith('core') || shortName.endsWith('-common')` to be "not end user facing"? - I mean should those maybe always have some sort of `internal` label that the script should enforce?

Reply:
don't think so, it' perfectly acceptable for people using core alone and internal is a little misleading i.e. right now if you want to have xml support you have to explicitly add xml-common